torch_xla/experimental/tuned_block_sizes.py

```diff
@@ -0,0 +1,66 @@
+def _next_power_of_2_bit_manipulation(x):
+  """
+    Finds the smallest power of 2 >= x using bit manipulation.
+    Assumes x is an integer.
+
+    Args:
+      x: The input number (should be an integer).
+
+    Returns:
+      The smallest integer power of 2 that is >= x.
+      Returns 1 if x <= 0.
+    """
+  if x <= 0:
+    return 1
+  if x == 1:
+    return 1
+  return 1 << (x - 1).bit_length()
+
+
+# ragged_paged_attention
+# key: (q_head_num, kv_head_num, token_num, max_model_len)
+# value: (num_kv_pages_per_block, num_queries_per_block)
+
+
+def _simplify_key_ragged_paged_attention(q_head_num, kv_head_num, token_num,
+                                         max_model_len):
+  token_num = _next_power_of_2_bit_manipulation(token_num)
+  max_model_len = _next_power_of_2_bit_manipulation(max_model_len)
+  return q_head_num, kv_head_num, token_num, max_model_len
+
+
+# TODO: add more tuned block sizes in the table
+_ragged_attention_table = {
+    (32, 8, 4096, 2048): (128, 64),
+    (4, 1, 4096, 2048): (128, 128),
+    (32, 8, 2048, 2048): (128, 32),
+    (4, 1, 2048, 2048): (128, 64),
+    (32, 8, 1024, 2048): (64, 32),
+    (1, 1, 1024, 2048): (64, 32),
+    (32, 8, 4096, 4096): (128, 64),
+    (4, 1, 4096, 4096): (128, 128),
+    (32, 8, 2048, 4096): (128, 32),
+    (4, 1, 2048, 4096): (128, 64),
+    (32, 8, 1024, 4096): (64, 32),
+    (1, 1, 1024, 4096): (64, 32),
+    (32, 8, 4096, 64): (32, 32),
+    (4, 1, 4096, 64): (32, 32),
+    (32, 8, 2048, 64): (32, 32),
+    (4, 1, 2048, 64): (32, 32),
+    (32, 8, 1024, 64): (32, 32),
+    (1, 1, 1024, 64): (32, 32),
+    (32, 8, 4096, 128): (32, 32),
+    (4, 1, 4096, 128): (32, 32),
+    (32, 8, 2048, 128): (32, 32),
```

Comment:
perhaps for future ideas, we also want _ragged_attention_table to depend on `kv_lens`, `cu_q_lens`, `num_seqs`, right? Maybe we can get some discrete sample and fit a curve

Reply:
Maybe, we can discuss with @bythew3i what's the best way to do that later.